Webpack 5 - platform and screen-size qualifier not usable at the same time anymore

### Issue Description

In one of our app we need different layouts for Android phones and tablets and iOS phone and tablets for some parts of the UI, so we have something like this:

- template.android.xml
- template.android.minWH600.xml
- template.ios.xml
- template.ios.minWH600.xml

While this was working as expected with @nativescript/core 8.1.5 and **@nativescript/webpack 4.1.0**, it is not correctly working anymore after upgrading to @nativescript/webpack 5.0.0.

When building the project for Android or iOS, I see the following warning:

> WARNING in ./app/ sync .(xml|js|s?css)$ ./snippets/test1.minWH600.xml
> Module not found: Error: Can't resolve './snippets/test1.minWH600.xml' in '/Users/felixk/Projects/ns-8-wp-5-test/app'
>  @ ./app/__@nativescript_webpack_virtual_entry_javascript__ 3:16-89

The result:
- [x] Android phone shows the correct layout
- [x] Android tablet shows the correct layout
- [x] iPhone shows the correct layout
- [ ] iPad shows the layout of Android tablet ??

I tried swapping the qualifiers to the following: _template.android.minWH600.xml_ -> _template.minWH600.android.xml_
The result is the same, except that I do not see the warning (provided above) during the build anymore.

----------------------------------
In another situation, we need to show a special layout only for iPads, so we have something like this:
- template.xml -> for Android phones & tablets + iPhone
- template.minWH600.ios.xml -> iPad

The result:
- [x] Android phone shows the correct layout
- [ ] Android tablet shows the layout of iPad
- [x] iPhone shows the correct layout
- [x] iPad shows the correct layout

### Reproduction

Sample app to reproduce:
[ns-8-wp-5-test.zip](https://github.com/NativeScript/NativeScript/files/7736596/ns-8-wp-5-test.zip)

### Relevant log output (if applicable)

_No response_

### Environment

Command does not work for some reaon, so here is the content of package.json:

```
{
  "name": "ns-8-wp-5-test",
  "main": "app/app.js",
  "version": "1.0.0",
  "private": true,
  "dependencies": {
    "@nativescript/core": "^8.1.5"
  },
  "devDependencies": {
    "@nativescript/android": "8.1.1",
    "@nativescript/ios": "8.1.0",
    "@nativescript/webpack": "~5.0.0"
  }
}
```

And the default webpack.config.json:

```
const webpack = require("@nativescript/webpack");

module.exports = (env) => {
	webpack.init(env);

	// Learn how to customize:
	// https://docs.nativescript.org/webpack

	return webpack.resolveConfig();
};
```


Ia am working with latest NativeScript CLI 8.1.5.

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

--- a/packages/core/module-name-resolver/qualifier-matcher/index.ts
+++ b/packages/core/module-name-resolver/qualifier-matcher/index.ts
@@ -119,20 +119,21 @@ const supportedQualifiers: Array<QualifierSpec> = [minWidthHeightQualifier, minW
 
 function checkQualifiers(path: string, context: PlatformContext): number {
 	let result = 0;
+	let value: number;
 	for (let i = 0; i < supportedQualifiers.length; i++) {
 		const qualifier = supportedQualifiers[i];
 		if (qualifier.isMatch(path)) {
 			const occurences = qualifier.getMatchOccurences(path);
 			// Always get the last qualifier among identical occurences
-			result = qualifier.getMatchValue(occurences[occurences.length - 1], context);
-			if (result < 0) {
+			value = qualifier.getMatchValue(occurences[occurences.length - 1], context);
+			if (value < 0) {
 				// Non of the supported qualifiers matched this or the match was not satisfied
 				return -1;
 			}
 
-			result += (supportedQualifiers.length - i) * PRIORITY_STEP;
-
-			return result;
+			if (value > 0) {
+				result += value + (supportedQualifiers.length - i) * PRIORITY_STEP;
+			}
 		}
 	}
 

